packages/core/src/types.ts

```diff
@@ -60,7 +68,7 @@ export interface Operation extends GraphQLRequest {
 }
 
 /** Resulting data from an [operation]{@link Operation}. */
-export interface OperationResult<Data = any> {
+export interface OperationResult<Data extends PrimitiveMap = PrimitiveMap> {
```

Comment:
```suggestion
export interface OperationResult<Data = PrimitiveMap | null> {
```

Let's make sure that we also take `null` into account. I'd either add it here or below to `data?: Data | null`, since it can always happen when the entire GraphQL endpoint errors out.

Also, I don't think we can add `extends PrimitiveMap` because the result may not be serialisable after it passed through graphcache or other custom exchanges that may add special data, so let's assume the user knows what they're doing when they pass in a `Data` generic that doesn't extend `PrimitiveMap`.

Also, I realised that users can get the "legacy behaviour" back by just passing `any` to the `Data` generic, so no harm done I guess.